File: Python3/프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    rows, cols = len(arr), len(arr[0])
    
    if rows > cols:
        for row in arr:
            row.extend([0] * (rows - cols))
    elif cols > rows:
        extra_row = [[0] * cols for _ in range(cols-rows)]
        arr.extend(extra_row)
        
    return arr

File: Swift/프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.swift
import Foundation

func solution(_ arr:[[Int]]) -> [[Int]] {
    var matrix = arr
    
    let rows = arr.count
    let cols = arr[0].count
    
    if rows > cols {
        for i in 0..<rows {
            matrix[i] += Array(repeating: 0, count: rows - cols)
        }
    } else if cols > rows {
        let extraRows = Array(repeating: Array(repeating: 0, count: cols), count: cols - rows)
        matrix.append(contentsOf: extraRows)
    }
    return matrix 
}